src/internals/rules.rs

```diff
@@ -114,6 +124,63 @@ impl<'a> From<(&'a yara_sys::YR_SCAN_CONTEXT, &'a yara_sys::YR_RULE)> for Rule<'
     }
 }
 
+impl<'a> From<(&'a yara_sys::YR_SCAN_CONTEXT, &'a yara_sys::YR_RULE)> for Rule<'a> {
+    fn from((context, rule): (&'a yara_sys::YR_SCAN_CONTEXT, &'a yara_sys::YR_RULE)) -> Self {
+        let mut result = Rule::from(rule);
+        result.strings = YrStringIterator::from(rule)
+            .map(|s| YrString::from((context, s)))
+            .collect();
+        result
+    }
+}
+
+/// Iterate over YR_RULE in a YR_RULES.
+///
+/// # Implementation notes
+///
+/// See `yr_rules_foreach` in Yara.
+pub struct RuleIterator<'a> {
+    head: *const yara_sys::YR_RULE,
+    _marker: marker::PhantomData<&'a yara_sys::YR_RULE>,
+}
+
+impl<'a> From<&'a yara_sys::YR_RULES> for RuleIterator<'a> {
+    fn from(rules: &'a yara_sys::YR_RULES) -> RuleIterator<'a> {
+        RuleIterator {
+            head: rules.get_rules_table(),
+            _marker: marker::PhantomData,
+        }
+    }
+}
+
+impl<'a> Iterator for RuleIterator<'a> {
+    type Item = RulesetRule<'a>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        if self.head.is_null() {
+            return None;
+        }
+
+        let rule = unsafe { *self.head };
+        let mut result: Option<Self::Item> = None;
+
+        if rule.flags & rule.flags == yara_sys::RULE_FLAGS_NULL {
```

Comment:
`rule.flags & rules.flags`? Did you meant `(rule.flags &  yara_sys::RULE_FLAGS_NULL) != 0`?

Reply:
Yes, good catch, fixed.